packages/headless/src/features/folding/folding-state.ts

```diff
@@ -1,12 +1,27 @@
 import {Result} from '../../api/search/search/result';
 
+export const collectionMoreResultsAvailableBuffer = 5;
```

Comment:
As mentioned in the previous PR, we want to include `moreResultsAvailable` with each collection so developers may display a "show more"/"show all" button only when pressing them does something. However, the API doesn't give us an indicator of whether more results are available, so the strategy used in this PR is to request `collectionMoreResultsAvailableBuffer` extra results, check if more results were in fact returned than requested, then trim them back before putting them in the state (to make sure there's a difference in the state after dispatching `loadCollection`).

In theory, given the above purpose, the buffer should only be `1`, but things get complicated.

### Problem 1: a parent can be a child too
As mentioned in the diagram, occasionally a `parentResult` may be redundantly added to the `childResults` array. In the worst case scenario (which I've witnessed happen), a collection may contain exactly as many results as set in `options.numberOfFoldedResults`, and requesting one extra result may add the `parentResult` to the `childResults` array, fooling our strategy into thinking there's more results available.

The solution for this would be to set the buffer to `2`, since there cannot be two different parents.

### Problem 2: there *can* be two different parents
In a hierarchy with more than two levels (where two levels could be a forum thread, for instance), there can be multiple different `parents` that all might be added to different `childResults` arrays. I believe `collectionMoreResultsAvailableBuffer` may only need to be equal to the maximum depth of the hierarchy.

### Solution
1. We attempt to guess a maximum depth that most indexes should respect, at a potential performance cost. 5 may be more than enough here. We could even choose to only officially support folding for a depth of 2 and set the buffer to 2.
2. We make it an option. This would most likely be very confusing for developers, I personally don't like it.
3. We abandon guessing whether more results are available altogether. Perhaps it could be a feature in the next search API version. This may be a bit annoying for users and developers since they may think the "loadCollection" feature is broken.
4. We put the entire "loadCollection"/"showMore" feature on hold.

Reply:
Thanks for describing the problem this way. It is helpful 💯 

It would be ideal if the api could return a `moreResultsAvailable` to avoid work-arounds in the client. cc: @acote-coveo 

Regarding the four solutions:
1. I feel this would be good enough in many cases but not all. When we have done this in the past, what I've seen is, at scale, the minority of cases end up being reported as maintenance cases. Personally, I prefer to avoid shipping features if they have somewhat common-to-hit limitations.
2. I agree with you, I do not think we should expose such an option.
3. I like this idea of doing less, but doing it well. Why would the "showMore" feature be considered broken in this case?
4. Could a 1-level deep showMore be a compromise between all or none?

Reply:
My personal opinion: Go with #3 . 

It is essentially what we have for JSUI, and we did not get too much "complaints" regarding this precisely about folding, all in all... 

I guess people are just happy when they get it working in the first place 😂